pkg/apis/sources/v1alpha1/ibmmq_types.go

```diff
@@ -53,19 +54,34 @@ type IBMMQSourceSpec struct {
 	Auth Credentials `json:"credentials"`
 }
 
-// Delivery defines the source's message delivery behavior
+// Delivery defines the source's message delivery behavior.
 type Delivery struct {
-	DeadLetterQueueManager string `json:"deadLetterQueueManager"`
-	DeadLetterQueue        string `json:"deadLetterQueue"`
-	Retry                  int    `json:"retry"`
+	DeadLetterQueue string `json:"deadLetterQueue"`
+	Retry           int    `json:"retry"`
+
 	// currently not used
-	BackoffDelay int `json:"backoffDelay"`
+	DeadLetterQueueManager string `json:"deadLetterQueueManager,omitempty"`
+	BackoffDelay           int    `json:"backoffDelay,omitempty"`
 }
 
-// Credentials holds the auth details
+// Credentials holds the auth details.
 type Credentials struct {
-	User     ValueFromField `json:"username"`
-	Password ValueFromField `json:"password"`
+	User     ValueFromField `json:"username,omitempty"`
+	Password ValueFromField `json:"password,omitempty"`
+	TLS      *TLSSpec       `json:"tls,omitempty"`
+}
+
+// TLSSpec holds the IBM MQ TLS authentication parameters.
+type TLSSpec struct {
+	Cipher             string   `json:"cipher"`
+	ClientAuthRequired bool     `json:"clientAuthRequired"`
+	KeyRepository      Keystore `json:"keyRepository"`
+}
+
+// Keystore represents Key Database components.
+type Keystore struct {
+	KeyDatabase   *corev1.SecretKeySelector `json:"kdbSecret"`
```

Comment:
This is neither compliant with the way targets nor sources have secrets exposed in specs, and we already have trouble agreeing on a common way across TriggerMesh, so we should at least stick to one of these two:

https://github.com/triggermesh/triggermesh/blob/6dc00197a0ea827840b501d1c9cd16ae536c1c5b/config/samples/targets/confluent/200-target.yaml#L30-L32

https://github.com/triggermesh/triggermesh/blob/6dc00197a0ea827840b501d1c9cd16ae536c1c5b/config/samples/sources/awscognitoidentitysource.yaml#L32-L34

Reply:
`valueFromSecret` added in 86af0b1